### PR TITLE
plugin/docker: Report on deploy status

### PR DESCRIPTION
**Pull request by briancain**
This commit introduces the implementation of the Status function for
docker on deploy. It queries the current deployed containers health and
looks for if it was configured using dockers built in health checks. If
not, the function attempts to make some basic assumptions for
determining the health of the container.

### CODE REVIEW

File: builtin/docker/platform.go [hunk 1]
```diff
@@ -95,6 +104,136 @@ func (p *Platform) resourceManager(log hclog.Logger) *resource.Manager {
...
+
+	// TODO(briancain): remove me GH #1469
+	time.Sleep(500 * time.Millisecond)
+
+	// update output based on main health state
+	s.Update("Finished building report for Docker platform")
+	s.Done()
+
+	st := ui.Status()
+	defer st.Close()
```
Comment (evanphx) [inline builtin/docker/platform.go:220]: Per the discussion last week, don't use Status, use the same stepgroup you have here and set the step's status.

File: builtin/docker/platform.go [hunk 2]
```diff
@@ -95,6 +104,136 @@ func (p *Platform) resourceManager(log hclog.Logger) *resource.Manager {
...
+	if result.Health == sdk.StatusReport_READY {
+		st.Step(terminal.StatusOK, fmt.Sprintf("Container %q is reporting ready!", containerInfo.Name))
+	} else if result.Health == sdk.StatusReport_PARTIAL {
+		st.Step(terminal.StatusWarn, fmt.Sprintf("Container %q is reporting partially available!", containerInfo.Name))
+	} else {
+		st.Step(terminal.StatusError, fmt.Sprintf("Container %q is reporting not ready!", containerInfo.Name))
+	}
+
+	// TODO(briancain): remove me GH #1469
+	time.Sleep(500 * time.Millisecond)
```
Comment (evanphx) [inline builtin/docker/platform.go:232]: Again, should be fixed, please retest after rebase!

**Review by evanphx — CHANGES_REQUESTED** [review]

File: builtin/docker/platform.go (hunk 1 above)
Comment (briancain) [reply builtin/docker/platform.go:220]: Ah yeah, switching back from Status to StepGroups, I remember now why I tried using the `ui.Status` directly. Every time I do something like this:

```golang
s.Update("some text to be printed!!")
s.Status(terminal.StatusOk)
```

The text shows up as just `ok` and not `"some text to be printed!!"`. Additionally if I do something like:

```golang
s.Update("some text to be printed!!")
s.Status(terminal.StatusWarn)
```

It just prints `warn` in a green status ok format.

File: builtin/docker/platform.go [hunk 3]
```diff
@@ -95,6 +104,136 @@ func (p *Platform) resourceManager(log hclog.Logger) *resource.Manager {
...
+		result.Health = sdk.StatusReport_ALIVE
+	} else {
+		result.Health = sdk.StatusReport_PARTIAL
+	}
+
+	result.TimeGenerated = ptypes.TimestampNow()
+	log.Debug("status report complete")
+
+	// TODO(briancain): remove me GH #1469
+	time.Sleep(500 * time.Millisecond)
```
[the remark this answers is not in the document]
Comment (briancain) [reply builtin/docker/platform.go]: It looks like it! Gave it a couple of tries and it printed right every time 🎉

File: builtin/docker/platform.go (hunk 1 above)
Comment (briancain) [reply builtin/docker/platform.go:216]: Quick note from our video sync: This code should be working as mentioned above, but the behavior described sounds like a bug. The right move is to still move away from Status, but we should address the output bug too.

**Comment by briancain** [comment]
Rebased off `main` now instead of the core status report impl. branch!

**Review by krantzinator — APPROVED** [review]
Woooo 🚀

**Pull request merged**